api/src/index.js

```diff
@@ -11,14 +11,70 @@ app.use(bodyParser.json());
 
 const apiRouter = express.Router();
 
-// You can delete this route once you add your own routes
-apiRouter.get("/", async (req, res) => {
-  const SHOW_TABLES_QUERY =
-    process.env.DB_CLIENT === "pg"
-      ? "SELECT * FROM pg_catalog.pg_tables;"
-      : "SHOW TABLES;";
-  const tables = await knex.raw(SHOW_TABLES_QUERY);
-  res.json({ tables });
+// Getting future meals
+apiRouter.get("/future-meals", async (req, res) => {
+	try {
+		const upcomingMeals = await knex.raw(
+			"SELECT * FROM meal WHERE `when` > NOW()"
+		);
+
+		upcomingMeals[0].length ?
+			res.json(upcomingMeals[0])
+		:	res.status(404).json("There are no any events in the future.");
+	} catch (error) {
+		res.status(500).json({ error: "Internal server error." });
+	}
+});
+
+//Getting past meals
+apiRouter.get("/past-meals", async (req, res) => {
+	try {
+		const pastMeals = await knex.raw("SELECT * FROM meal WHERE `when` < NOW()");
+
+		pastMeals[0].length ?
+			res.json(pastMeals[0])
+		:	res.status(404).json("There are no any past events.");
+	} catch (error) {
+		res.status(500).json({ error: "Internal server error." });
+	}
+});
+
+//Getting all meals
+apiRouter.get("/all-meals", async (req, res) => {
+	try {
+		const allMeals = await knex.raw("SELECT * FROM meal ORDER BY id");
+		allMeals[0].length ?
+			res.json(allMeals[0])
+		:	res.status(404).json("Meals not found.");
+	} catch (error) {
+		res.status(500).json({ error: "Internal server error." });
+	}
+});
+
+//Getting first meal
+apiRouter.get("/first-meal", async (req, res) => {
+	try {
+		const firstMeal = await knex.raw("SELECT * FROM meal ORDER BY id LIMIT 1");
+		firstMeal[0].length ?
+			res.json(firstMeal[0])
```

Comment:
The `firstMeal[0]` variable is an array. As the assignment asked, in cases where we only expect a single result, the returned value should be an **object**, not an **array**. Thos means you should write `firstMeal[0][0]` instead.

This goes back to my previous comment, that having a more explaining variable would make it clearer what the code does.